karton/core/backend/base.py

```diff
@@ -42,6 +44,35 @@ class KartonServiceInfo:
     service_version: str | None = None
 
 
+def resolve_service_info(
+    identity: str | None, service_info: KartonServiceInfo | None
+) -> KartonServiceInfo:
+    """
+    Makes KartonServiceInfo object from backend parameters that identify the service.
+
+    If only identity is provided: KartonServiceInfo containing only an identity and
+    karton_version information is created.
+
+    This function also validates if identity is correct and doesn't contain disallowed
+    characters.
+    """
+    if service_info is None:
+        if identity is None:
+            raise InvalidIdentityError("Identity cannot be None")
```

Comment:
Maybe something like:

```suggestion
            raise InvalidIdentityError("Either identity or service_info must be provided")
```